### ISSUE TITLE
Nuxt 2 + Vuelidate 2 +vuejs2.7.10 not working

### ISSUE THREAD

**Issue by rdominl2003** [vuelidate/vuelidate#1101]
### Make sure that you are familiar with documentation before submitting an issue.

  https://vuelidate-next.netlify.com/

**Describe the bug**
I was able to create a form using **Vuelidate 2 + vuejs2.6.14 + composition-api.** But after **Vuejs 2.7** was released, I have tried to stop using composition-api lib and use the reactive from vue without any luck.
It's feels like the reactivity is not working as it was before when I was using composition-api lib.

**Reproduction URL**
Here I drop a repo https://github.com/rdominl2003/vuelidate where I made a small example, using what's in the getting started page of the documentation https://vuelidate-next.netlify.com/.
Basically what I'm targeting is **(Vuelidate 2 - Composition API + Vue 2.7.x**)

**Expected behavior**
That form + validations behave the same that (Vuelidate 2 - Composition API + Vuejs 2.6.14)

**Comment by rdominl2003** [comment]
@dobromir-hristov Does any of you guys can take a look into and help me out with this?
Thanks in advance

**Comment by dargmuesli** [comment]
Maybe describe the actual issue you're facing, that way people know that the issue is about.

**Comment by rdominl2003** [comment]
@dargmuesli happens that $model is always undefined, no matter what approach you take to try to update the model.

**Comment by vaidd4** [comment]
I have been trying to make vuelidate 2 work with nuxt 2 for some time now but without success.
I have managed to make a minimal reproducible project : https://stackblitz.com/edit/nuxt-typescript-starter-na9sw4?file=components/vuelidate.vue
( `nuxt@2.15.8`, `vue@2.7.14`, `@vuelidate/{core,validators}@2.0.0` )

I have tried many different things: 
- `ref(...)` instead of `reactive(...)`
- Rules in `computed(() => ...)`
- `useVuelidate()` + `validations()` instead of composition
  (vuelidate can't seam to find the `validations` method in the component, seams related to https://github.com/vuelidate/vuelidate/issues/1099)
- Using setup method with Typescript Class Component (`@Component({ setup() {...} })`
- Using `beforeMounted` hook instead of setup method
- and more ...

But it seams to come down to Nuxt, something in Nuxt behavior prevent the `$model` to update.
I tested the same component in a Vue@2.7.14 only project and it works as expected

A weird behaviour that might help understand the issue: 
Try replacing input by `<input v-model="v.state.text.$model" />`, you can see that $model can't be updated, as if it was reset. 

Edit: Following what @rdominl2003 said, I tried to rollback to `vue@2.6.4` in the same reproducible projet and it worked!
Might be linked to `@vue/composition-api` or `vue-demi` packages?
https://stackblitz.com/edit/nuxt-typescript-starter-2ifeq4?file=components%2Fvuelidate.vue
(had to run `npx vue-demi-switch 2` in the stackblitz console)

**Comment by vaidd4** [comment]
I have somewhat of a workaround:
By modifying directly `node_modules/@vuelidate/core/dist/index.esm.js` file and changing the import `from 'vue-demi'` to `from 'vue'` the vuelidate library seems to work as expected.
(option `useVuelidate() + validations()` or composition `useVuelidate(rules, state)` alike)

It seems that vue-demi doesn't play well with Nuxt 2 and vue 2.7 but I have still no idea why.

May be this is related, idk: https://blog.vuejs.org/posts/vue-2-7-naruto.html#behavior-differences-from-vue-3
> `reactive()`, `ref()`, and `shallowReactive()` will directly convert original objects instead of creating proxies.

**Comment by Leander250** [comment]
> I have somewhat of a workaround: By modifying directly `node_modules/@vuelidate/core/dist/index.esm.js` file and changing the import `from 'vue-demi'` to `from 'vue'` the vuelidate library seems to work as expected. (option `useVuelidate() + validations()` or composition `useVuelidate(rules, state)` alike)
> 
> It seems that vue-demi doesn't play well with Nuxt 2 and vue 2.7 but I have still no idea why.
> 
> May be this is related, idk: https://blog.vuejs.org/posts/vue-2-7-naruto.html#behavior-differences-from-vue-3
> 
> > `reactive()`, `ref()`, and `shallowReactive()` will directly convert original objects instead of creating proxies.

Thanks that made it finally work after a few hours of searching for a solution!


    "@vuelidate/core": "^2.0.0",
    "@vuelidate/validators": "^2.0.0",
    "nuxt": "^2.16.2",
    "vue": "2.7.14",

**Comment by shentao** [comment]
When upgrading to Vue 2.7, you sometimes need to delete your package-lock and node_modules and perform a clean `npm install`.
